pkg/datamodel-tests/src/tests.ts

```diff
@@ -389,3 +389,53 @@ export const outNotesStream: TestFunction = async (db) => {
 		expect(outNoteList).toEqual([{ id: note1._id, displayName: 'New Note - Updated' }]);
 	});
 };
+
+export const sequenceWarehouseDesignDocument: TestFunction = async (db) => {
+	const wh1 = await db.warehouse('0000').create(); // New Warehouse
+	const wh2 = await db.warehouse('1').create(); // New Warehouse (1)
+	const wh3 = await db.warehouse('2').create(); // New Warehouse (2)
+
+	expect(wh1.displayName).toEqual('New Warehouse');
+	expect(wh2.displayName).toEqual('New Warehouse (1)');
+	expect(wh3.displayName).toEqual('New Warehouse (2)');
+
+	wh1.setName('New Name1');
+	wh2.setName('New Name2');
+
+	const wh4 = await db.warehouse('3').create(); // New Warehouse (3)
+	expect(wh4.displayName).toEqual('New Warehouse (3)');
+
+	wh3.setName('New Name3');
+	wh4.setName('New Name4');
+
+	const wh5 = await db.warehouse('4').create(); // New Warehouse
+	expect(wh5.displayName).toEqual('New Warehouse');
+};
+
+export const sequenceNoteDesignDocument: TestFunction = async (db) => {
+	const defaultWarehouse = await db.warehouse().create();
+
+	const note1 = await defaultWarehouse.note().create(); // New Note
+	const note1FromDB = await defaultWarehouse.note(note1._id).get();
+	const note2 = await defaultWarehouse.note().create(); // New Note (1)
+	const note2FromDB = await defaultWarehouse.note(note2._id).get();
+	const note3 = await defaultWarehouse.note().create(); // New Note (2)
+	const note3FromDB = await defaultWarehouse.note(note3._id).get();
+
+	expect(note1FromDB).toMatchObject({ displayName: 'New Note' });
+	expect(note2FromDB).toMatchObject({ displayName: 'New Note (1)' });
```

Comment:
Yeah, I think that one time I went for `ObjectContaining` was because it asserts for nested objects too, while `toMatchObject` doesn't. 
https://stackoverflow.com/questions/45692456/whats-the-difference-between-tomatchobject-and-objectcontaining